[telegram] Installation via PaperUI fails with exceptions

When I try to install the Telegram binding via the PaperUI, I see a lot of exceptions in the karaf console. Compiling via maven and using the JAR works however.

```
openhab> org.apache.karaf.features.internal.util.MultiException: Error:
        Error downloading wrap:file:/C:/Users/winuser/DOWNLO~1/OPENHA~1.M6/userdata/tmp/mvn/com/squareup/okio/okio/1.15.0/okio-1.15.0.jar
        Error downloading wrap:file:/C:/Users/winuser/DOWNLO~1/OPENHA~1.M6/userdata/tmp/mvn/com/squareup/okio/okio/1.15.0/okio-1.15.0.jar
        Error downloading wrap:file:/C:/Users/winuser/DOWNLO~1/OPENHA~1.M6/userdata/tmp/mvn/com/squareup/okhttp3/logging-interceptor/3.12.3/logging-interceptor-3.12.3.jar
        Error downloading wrap:file:/C:/Users/winuser/DOWNLO~1/OPENHA~1.M6/userdata/tmp/mvn/com/squareup/okhttp3/logging-interceptor/3.12.3/logging-interceptor-3.12.3.jar
        Error downloading wrap:file:/C:/Users/winuser/DOWNLO~1/OPENHA~1.M6/userdata/tmp/mvn/com/github/pengrad/java-telegram-bot-api/4.4.0/java-telegram-bot-api-4.4.0.jar
        Error downloading wrap:file:/C:/Users/winuser/DOWNLO~1/OPENHA~1.M6/userdata/tmp/mvn/com/github/pengrad/java-telegram-bot-api/4.4.0/java-telegram-bot-api-4.4.0.jar
        Error downloading wrap:file:/C:/Users/winuser/DOWNLO~1/OPENHA~1.M6/userdata/tmp/mvn/com/squareup/okhttp3/okhttp/3.12.3/okhttp-3.12.3.jar
        Error downloading wrap:file:/C:/Users/winuser/DOWNLO~1/OPENHA~1.M6/userdata/tmp/mvn/com/squareup/okhttp3/okhttp/3.12.3/okhttp-3.12.3.jar
        at org.apache.karaf.features.internal.download.impl.MavenDownloadManager$MavenDownloader.<init>(MavenDownloadManager.java:91)
        at org.apache.karaf.features.internal.download.impl.MavenDownloadManager.createDownloader(MavenDownloadManager.java:72)
        at org.apache.karaf.features.internal.region.Subsystem.downloadBundles(Subsystem.java:457)
        at org.apache.karaf.features.internal.region.Subsystem.downloadBundles(Subsystem.java:452)
        at org.apache.karaf.features.internal.region.SubsystemResolver.resolve(SubsystemResolver.java:224)
        at org.apache.karaf.features.internal.service.Deployer.deploy(Deployer.java:393)
        at org.apache.karaf.features.internal.service.FeaturesServiceImpl.doProvision(FeaturesServiceImpl.java:1062)
        at org.apache.karaf.features.internal.service.FeaturesServiceImpl.lambda$doProvisionInThread$13(FeaturesServiceImpl.java:998)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
        Suppressed: java.io.IOException: Error downloading wrap:file:/C:/Users/winuser/DOWNLO~1/OPENHA~1.M6/userdata/tmp/mvn/com/squareup/okio/okio/1.15.0/okio-1.15.0.jar
                at org.apache.karaf.features.internal.download.impl.AbstractRetryableDownloadTask.run(AbstractRetryableDownloadTask.java:77)
                at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
                at java.util.concurrent.FutureTask.run(FutureTask.java:266)
                at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
                at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
                ... 3 more
        Caused by: java.io.IOException: Could not download [wrap:file:/C:/Users/winuser/DOWNLO~1/OPENHA~1.M6/userdata/tmp/mvn/com/squareup/okio/okio/1.15.0/okio-1.15.0.jar]
                at org.apache.karaf.features.internal.download.impl.SimpleDownloadTask.download(SimpleDownloadTask.java:92)
                at org.apache.karaf.features.internal.download.impl.AbstractRetryableDownloadTask.run(AbstractRetryableDownloadTask.java:60)
                ... 7 more
        Caused by: java.net.MalformedURLException: unknown protocol: wrap
                at java.net.URL.<init>(URL.java:600)
                at java.net.URL.<init>(URL.java:490)
                at java.net.URL.<init>(URL.java:439)
                at org.apache.karaf.features.internal.download.impl.SimpleDownloadTask.download(SimpleDownloadTask.java:63)
                ... 8 more
        [CIRCULAR REFERENCE:java.io.IOException: Error downloading wrap:file:/C:/Users/winuser/DOWNLO~1/OPENHA~1.M6/userdata/tmp/mvn/com/squareup/okio/okio/1.15.0/okio-1.15.0.jar]
        Suppressed: java.io.IOException: Error downloading wrap:file:/C:/Users/winuser/DOWNLO~1/OPENHA~1.M6/userdata/tmp/mvn/com/squareup/okhttp3/logging-interceptor/3.12.3/logging-interceptor-3.12.3.jar
                at org.apache.karaf.features.internal.download.impl.AbstractRetryableDownloadTask.run(AbstractRetryableDownloadTask.java:77)
                at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
                at java.util.concurrent.FutureTask.run(FutureTask.java:266)
                at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
                at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
                ... 3 more
        Caused by: java.io.IOException: Could not download [wrap:file:/C:/Users/winuser/DOWNLO~1/OPENHA~1.M6/userdata/tmp/mvn/com/squareup/okhttp3/logging-interceptor/3.12.3/logging-interceptor-3.12.3.jar]
                at org.apache.karaf.features.internal.download.impl.SimpleDownloadTask.download(SimpleDownloadTask.java:92)
                at org.apache.karaf.features.internal.download.impl.AbstractRetryableDownloadTask.run(AbstractRetryableDownloadTask.java:60)
                ... 7 more
        Caused by: java.net.MalformedURLException: unknown protocol: wrap
                at java.net.URL.<init>(URL.java:600)
                at java.net.URL.<init>(URL.java:490)
                at java.net.URL.<init>(URL.java:439)
                at org.apache.karaf.features.internal.download.impl.SimpleDownloadTask.download(SimpleDownloadTask.java:63)
                ... 8 more
        [CIRCULAR REFERENCE:java.io.IOException: Error downloading wrap:file:/C:/Users/winuser/DOWNLO~1/OPENHA~1.M6/userdata/tmp/mvn/com/squareup/okhttp3/logging-interceptor/3.12.3/logging-interceptor-3.12.3.jar]
        Suppressed: java.io.IOException: Error downloading wrap:file:/C:/Users/winuser/DOWNLO~1/OPENHA~1.M6/userdata/tmp/mvn/com/github/pengrad/java-telegram-bot-api/4.4.0/java-telegram-bot-api-4.4.0.jar
                at org.apache.karaf.features.internal.download.impl.AbstractRetryableDownloadTask.run(AbstractRetryableDownloadTask.java:77)
                at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
                at java.util.concurrent.FutureTask.run(FutureTask.java:266)
                at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
                at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
                ... 3 more
        Caused by: java.io.IOException: Could not download [wrap:file:/C:/Users/winuser/DOWNLO~1/OPENHA~1.M6/userdata/tmp/mvn/com/github/pengrad/java-telegram-bot-api/4.4.0/java-telegram-bot-api-4.4.0.jar]
                at org.apache.karaf.features.internal.download.impl.SimpleDownloadTask.download(SimpleDownloadTask.java:92)
                at org.apache.karaf.features.internal.download.impl.AbstractRetryableDownloadTask.run(AbstractRetryableDownloadTask.java:60)
                ... 7 more
        Caused by: java.net.MalformedURLException: unknown protocol: wrap
                at java.net.URL.<init>(URL.java:600)
                at java.net.URL.<init>(URL.java:490)
                at java.net.URL.<init>(URL.java:439)
                at org.apache.karaf.features.internal.download.impl.SimpleDownloadTask.download(SimpleDownloadTask.java:63)
                ... 8 more
        [CIRCULAR REFERENCE:java.io.IOException: Error downloading wrap:file:/C:/Users/winuser/DOWNLO~1/OPENHA~1.M6/userdata/tmp/mvn/com/github/pengrad/java-telegram-bot-api/4.4.0/java-telegram-bot-api-4.4.0.jar]
        Suppressed: java.io.IOException: Error downloading wrap:file:/C:/Users/winuser/DOWNLO~1/OPENHA~1.M6/userdata/tmp/mvn/com/squareup/okhttp3/okhttp/3.12.3/okhttp-3.12.3.jar
                at org.apache.karaf.features.internal.download.impl.AbstractRetryableDownloadTask.run(AbstractRetryableDownloadTask.java:77)
                at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
                at java.util.concurrent.FutureTask.run(FutureTask.java:266)
                at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
                at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
                ... 3 more
        Caused by: java.io.IOException: Could not download [wrap:file:/C:/Users/winuser/DOWNLO~1/OPENHA~1.M6/userdata/tmp/mvn/com/squareup/okhttp3/okhttp/3.12.3/okhttp-3.12.3.jar]
                at org.apache.karaf.features.internal.download.impl.SimpleDownloadTask.download(SimpleDownloadTask.java:92)
                at org.apache.karaf.features.internal.download.impl.AbstractRetryableDownloadTask.run(AbstractRetryableDownloadTask.java:60)
                ... 7 more
        Caused by: java.net.MalformedURLException: unknown protocol: wrap
                at java.net.URL.<init>(URL.java:600)
                at java.net.URL.<init>(URL.java:490)
                at java.net.URL.<init>(URL.java:439)
                at org.apache.karaf.features.internal.download.impl.SimpleDownloadTask.download(SimpleDownloadTask.java:63)
                ... 8 more
        [CIRCULAR REFERENCE:java.io.IOException: Error downloading wrap:file:/C:/Users/winuser/DOWNLO~1/OPENHA~1.M6/userdata/tmp/mvn/com/squareup/okhttp3/okhttp/3.12.3/okhttp-3.12.3.jar]
```

## Steps to Reproduce (for Bugs)
1. Click the install button in PaperUI

## Your Environment
<!-- Include as many relevant details about the environment you experienced the bug in -->
* Version used: OH 2.5.0.M6
* Environment name and version: openjdk version "1.8.0_222",OpenJDK Runtime Environment (Zulu 8.40.0.25-win64)-Microsoft-Azure-restricted (build 1.8.0_222-b10),OpenJDK 64-Bit Server VM (Zulu 8.40.0.25-win64)-Microsoft-Azure-restricted (build 25.222-b10, mixed mode)
* Operating System and version: Windows 64 bit

--- a/bundles/org.openhab.binding.telegram/pom.xml
+++ b/bundles/org.openhab.binding.telegram/pom.xml
@@ -41,6 +41,7 @@
         <groupId>com.squareup.okhttp3</groupId>
         <artifactId>logging-interceptor</artifactId>
         <version>3.12.3</version>
+        <scope>compile</scope>
     </dependency>
   </dependencies>
 </project>

--- a/bundles/org.openhab.binding.telegram/src/main/feature/feature.xml
+++ b/bundles/org.openhab.binding.telegram/src/main/feature/feature.xml
@@ -5,9 +5,5 @@
     <feature name="openhab-binding-telegram" description="Telegram Binding" version="${project.version}">
         <feature>openhab-runtime-base</feature>
         <bundle start-level="80">mvn:org.openhab.addons.bundles/org.openhab.binding.telegram/${project.version}</bundle>
-        <bundle start-level="80">wrap:mvn:com.github.pengrad/java-telegram-bot-api/4.4.0</bundle>
-        <bundle start-level="80">wrap:mvn:com.squareup.okhttp3/okhttp/3.12.3</bundle>
-        <bundle start-level="80">wrap:mvn:com.squareup.okio/okio/1.15.0</bundle>
-        <bundle start-level="80">wrap:mvn:com.squareup.okhttp3/logging-interceptor/3.12.3</bundle>        
     </feature>
 </features>

--- a/bundles/org.openhab.binding.telegram/src/main/java/org/openhab/binding/telegram/bot/TelegramActions.java
+++ b/bundles/org.openhab.binding.telegram/src/main/java/org/openhab/binding/telegram/bot/TelegramActions.java
@@ -75,6 +75,7 @@ public class TelegramActions implements ThingActions {
         return true;
     }
 
+    @NonNullByDefault
     private static class BasicResult implements Authentication.Result {
 
         private final HttpHeader header;
@@ -142,7 +143,6 @@ public class TelegramActions implements ThingActions {
                 return false;
             }
             return message != null ? sendTelegram(chatId, message) : true;
-
         }
         return false;
     }

--- a/bundles/org.openhab.binding.telegram/src/main/java/org/openhab/binding/telegram/internal/TelegramConfiguration.java
+++ b/bundles/org.openhab.binding.telegram/src/main/java/org/openhab/binding/telegram/internal/TelegramConfiguration.java
@@ -12,6 +12,9 @@
  */
 package org.openhab.binding.telegram.internal;
 
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
 import java.util.List;
 
 /**
@@ -19,24 +22,26 @@ import java.util.List;
  *
  * @author Jens Runge - Initial contribution
  */
+@NonNullByDefault
 public class TelegramConfiguration {
 
     /**
      * Sample configuration parameter. Replace with your own.
      */
-    private String botUsername, botToken;
-    private List<String> chatIds;
-    private String parseMode;
+    private @Nullable String botUsername;
+    private @Nullable String botToken;
+    private @Nullable List<String> chatIds;
+    private String parseMode = "";
 
-    public String getBotUsername() {
+    public @Nullable String getBotUsername() {
         return botUsername;
     }
 
-    public String getBotToken() {
+    public @Nullable String getBotToken() {
         return botToken;
     }
 
-    public List<String> getChatIds() {
+    public @Nullable List<String> getChatIds() {
         return chatIds;
     }
 

--- a/bundles/org.openhab.binding.telegram/src/main/java/org/openhab/binding/telegram/internal/TelegramHandler.java
+++ b/bundles/org.openhab.binding.telegram/src/main/java/org/openhab/binding/telegram/internal/TelegramHandler.java
@@ -62,6 +62,7 @@ import okhttp3.OkHttpClient;
 @NonNullByDefault
 public class TelegramHandler extends BaseThingHandler {
 
+    @NonNullByDefault
     private class ReplyKey {
 
         final Long chatId;

--- a/bundles/org.openhab.binding.telegram/src/main/resources/ESH-INF/thing/thing-types.xml
+++ b/bundles/org.openhab.binding.telegram/src/main/resources/ESH-INF/thing/thing-types.xml
@@ -23,8 +23,8 @@
 				<description>Enter the bot token you received from the "BotFather".</description>
 			</parameter>
 			<parameter name="chatIds" type="text" required="true" multiple="true">
-				<label>Chat Ids</label>
-				<description>Enter your chat id. Only messages from this id will be send to openHAB.</description>
+				<label>Chat Id(s)</label>
+				<description>Enter your chat id(s). Only messages from this id(s) will be send to openHAB.</description>
 			</parameter>
 			<parameter name="parseMode" type="text" required="false">
 				<label>Parse Mode</label>
@@ -33,10 +33,10 @@
 					<option value="HTML">HTML</option>
 					<option value="Markdown">Markdown</option>
 				</options>
+				<default></default>
 				<description>Support for formatted messages, values: Markdown or HTML. Default: no formatting is used.</description>
 			</parameter>
 		</config-description>
-
 	</thing-type>
 
 	<channel-type id="lastMessageText">

